sympy/matrices/expressions/tests/test_derivatives.py

```diff
@@ -246,9 +250,23 @@ def test_derivatives_of_complicated_matrix_expr():
 
 def test_mixed_deriv_mixed_expressions():
 
+    expr = 3*Trace(A)
+    assert expr.diff(A) == 3*Identity(k)
+
+    expr = k
+    deriv = expr.diff(A)
+    assert isinstance(deriv, ZeroMatrix)
```

Comment:
Is this not tested by the line below?

Reply:
There are cases in which `==` returns `True` even if the objects do not belong to the same class.